django/contrib/gis/geos/geometry.py

```diff
@@ -496,6 +496,46 @@ def buffer(self, width, quadsegs=8):
         """
         return self._topology(capi.geos_buffer(self.ptr, width, quadsegs))
 
+    def buffer_with_style(self, width, quadsegs=8, end_cap_style=1, join_style=1, mitre_limit=5.0):
+        """
+        Advanced version of the `self.buffer()`. Does the same, but allows to customize
+        the styles of the final value.
+
+        [!] Needs GEOS-3.2 or higher
+
+        About styles: http://www.postgis.net/docs/ST_Buffer.html
+
+        :type width: float
```

Comment:
Please try to follow the style of existing docstrings -- we don't want all the type annotation stuff.